### ISSUE TITLE
Detect if model has been trained on the benchmarks

### ISSUE THREAD

**Issue by Mikeriess** [ScandEval/ScandEval#467]
### 🚀 The feature, motivation and pitch

The idea here is to use the approach in this paper:
https://arxiv.org/pdf/2406.06443

If possible, to detect if a model has been trained on any of the benchmark datasets. I have only read the summary of the paper so I dont know how demanding this is, but this could be a tool in the toolbox in the case there is suspicion that a model has been trained on one or more of the evaluation datasets.

**Comment by saattrupdan** [comment]
I like the idea of testing what the chance is for a model to have "cheated" on a benchmark. However, the method in the paper that you link to requires, for each evaluation dataset, to have a hidden split that we _know_ the model wasn't trained on. The problem is that we don't really have that. They also show that it's not enough to have a single hidden dataset, it has to come from the same distribution as the dataset you're testing. 

I'm not sure if things like perplexity could be used instead. Mean perplexity across the test set, say? It would require testing to say for sure. 

Also, such methods are for decoders only. That's in principle fine (we could use NaN values for encoders), but a method that works for both would be great. Maybe we could use pseudo-perplexity there.

**Comment by Mikeriess** [comment]
Sounds like a paper to me :-D Thanks for taking a look. I think your idea with perplexity makes good sense to use in this context. Sounds similar to the MIA-approaches mentioned in the paper - not the definite truth, but might be useful for flagging suspicion?

In any case we can keep our eyes open for this stream of research.

**Comment by DavidicLineage** [comment]
Seems to me that unless the model is built from the outset to establish and retain provenance, this is fundamentally a ground-truth problem. Any post-hoc detector is inferring provenance from model behaviour, so it may flag suspicious cases, but you can never know whether you've caught all the positives—or establish the false-negative rate—without independently known training data.
